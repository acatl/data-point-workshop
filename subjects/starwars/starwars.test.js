/* eslint-env jest */

const starwars = require('./starwars')

test('should request all people', function () {
  return starwars.getPeople()
    .then(acc => {
      expect(acc.value).toMatchSnapshot()
    })
})

test('should request luke', function () {
  return starwars.getLuke()
    .then(acc => {
      expect(acc.value).toMatchSnapshot()
    })
})

test('should get lukes basic info', function () {
  return starwars.getLukeBasicInfo()
    .then(acc => {
      expect(acc.value).toMatchSnapshot()
    })
})

test('should get luke details with species info', function () {
  return starwars.getLukeSpeciesInfo()
    .then(acc => {
      expect(acc.value).toMatchSnapshot()
    })
})

test('should get lukes details with starships', function () {
  return starwars.getLukeStarshipDetails()
    .then(acc => {
      expect(acc.value).toMatchSnapshot()
    })
})

test('should get lukes details with starships names', function () {
  return starwars.getLukeStarshipNames()
    .then(acc => {
      expect(acc.value).toMatchSnapshot()
    })
})

test('should get all people', function () {
  return starwars.getAllPeople()
    .then(acc => {
      expect(acc.value).toMatchSnapshot()
    })
})
