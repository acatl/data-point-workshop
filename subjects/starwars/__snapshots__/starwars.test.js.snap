// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get all people 1`] = `
Array [
  Object {
    "birthYear": "19BBY",
    "name": "Luke Skywalker",
    "species": Object {
      "classification": "mammal",
      "language": "Galactic Basic",
      "name": "Human",
    },
    "starships": Array [
      "X-wing",
      "Imperial shuttle",
    ],
  },
  Object {
    "birthYear": "112BBY",
    "name": "C-3PO",
    "species": Object {
      "classification": "artificial",
      "language": "n/a",
      "name": "Droid",
    },
    "starships": Array [],
  },
  Object {
    "birthYear": "33BBY",
    "name": "R2-D2",
    "species": Object {
      "classification": "artificial",
      "language": "n/a",
      "name": "Droid",
    },
    "starships": Array [],
  },
  Object {
    "birthYear": "41.9BBY",
    "name": "Darth Vader",
    "species": Object {
      "classification": "mammal",
      "language": "Galactic Basic",
      "name": "Human",
    },
    "starships": Array [
      "TIE Advanced x1",
    ],
  },
  Object {
    "birthYear": "19BBY",
    "name": "Leia Organa",
    "species": Object {
      "classification": "mammal",
      "language": "Galactic Basic",
      "name": "Human",
    },
    "starships": Array [],
  },
  Object {
    "birthYear": "52BBY",
    "name": "Owen Lars",
    "species": Object {
      "classification": "mammal",
      "language": "Galactic Basic",
      "name": "Human",
    },
    "starships": Array [],
  },
  Object {
    "birthYear": "47BBY",
    "name": "Beru Whitesun lars",
    "species": Object {
      "classification": "mammal",
      "language": "Galactic Basic",
      "name": "Human",
    },
    "starships": Array [],
  },
  Object {
    "birthYear": "unknown",
    "name": "R5-D4",
    "species": Object {
      "classification": "artificial",
      "language": "n/a",
      "name": "Droid",
    },
    "starships": Array [],
  },
  Object {
    "birthYear": "24BBY",
    "name": "Biggs Darklighter",
    "species": Object {
      "classification": "mammal",
      "language": "Galactic Basic",
      "name": "Human",
    },
    "starships": Array [
      "X-wing",
    ],
  },
  Object {
    "birthYear": "57BBY",
    "name": "Obi-Wan Kenobi",
    "species": Object {
      "classification": "mammal",
      "language": "Galactic Basic",
      "name": "Human",
    },
    "starships": Array [
      "Jedi starfighter",
      "Trade Federation cruiser",
      "Naboo star skiff",
      "Jedi Interceptor",
      "Belbullab-22 starfighter",
    ],
  },
]
`;

exports[`should get luke details with species info 1`] = `
Object {
  "classification": "mammal",
  "language": "Galactic Basic",
  "name": "Human",
}
`;

exports[`should get lukes basic info 1`] = `
Object {
  "birthYear": "19BBY",
  "name": "Luke Skywalker",
  "species": Array [
    "https://swapi.co/api/species/1/",
  ],
  "starships": Array [
    "https://swapi.co/api/starships/12/",
    "https://swapi.co/api/starships/22/",
  ],
}
`;

exports[`should get lukes details with starships 1`] = `
Array [
  Object {
    "MGLT": "100",
    "cargo_capacity": "110",
    "consumables": "1 week",
    "cost_in_credits": "149999",
    "created": "2014-12-12T11:19:05.340000Z",
    "crew": "1",
    "edited": "2014-12-22T17:35:44.491233Z",
    "films": Array [
      "https://swapi.co/api/films/2/",
      "https://swapi.co/api/films/3/",
      "https://swapi.co/api/films/1/",
    ],
    "hyperdrive_rating": "1.0",
    "length": "12.5",
    "manufacturer": "Incom Corporation",
    "max_atmosphering_speed": "1050",
    "model": "T-65 X-wing",
    "name": "X-wing",
    "passengers": "0",
    "pilots": Array [
      "https://swapi.co/api/people/1/",
      "https://swapi.co/api/people/9/",
      "https://swapi.co/api/people/18/",
      "https://swapi.co/api/people/19/",
    ],
    "starship_class": "Starfighter",
    "url": "https://swapi.co/api/starships/12/",
  },
  Object {
    "MGLT": "50",
    "cargo_capacity": "80000",
    "consumables": "2 months",
    "cost_in_credits": "240000",
    "created": "2014-12-15T13:04:47.235000Z",
    "crew": "6",
    "edited": "2014-12-22T17:35:44.795405Z",
    "films": Array [
      "https://swapi.co/api/films/2/",
      "https://swapi.co/api/films/3/",
    ],
    "hyperdrive_rating": "1.0",
    "length": "20",
    "manufacturer": "Sienar Fleet Systems",
    "max_atmosphering_speed": "850",
    "model": "Lambda-class T-4a shuttle",
    "name": "Imperial shuttle",
    "passengers": "20",
    "pilots": Array [
      "https://swapi.co/api/people/1/",
      "https://swapi.co/api/people/13/",
      "https://swapi.co/api/people/14/",
    ],
    "starship_class": "Armed government transport",
    "url": "https://swapi.co/api/starships/22/",
  },
]
`;

exports[`should get lukes details with starships names 1`] = `
Array [
  "X-wing",
  "Imperial shuttle",
]
`;

exports[`should request all people 1`] = `
Object {
  "count": 87,
  "next": "https://swapi.co/api/people/?page=2",
  "previous": null,
  "results": Array [
    Object {
      "birth_year": "19BBY",
      "created": "2014-12-09T13:50:51.644000Z",
      "edited": "2014-12-20T21:17:56.891000Z",
      "eye_color": "blue",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
        "https://swapi.co/api/films/7/",
      ],
      "gender": "male",
      "hair_color": "blond",
      "height": "172",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "77",
      "name": "Luke Skywalker",
      "skin_color": "fair",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [
        "https://swapi.co/api/starships/12/",
        "https://swapi.co/api/starships/22/",
      ],
      "url": "https://swapi.co/api/people/1/",
      "vehicles": Array [
        "https://swapi.co/api/vehicles/14/",
        "https://swapi.co/api/vehicles/30/",
      ],
    },
    Object {
      "birth_year": "112BBY",
      "created": "2014-12-10T15:10:51.357000Z",
      "edited": "2014-12-20T21:17:50.309000Z",
      "eye_color": "yellow",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/5/",
        "https://swapi.co/api/films/4/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "n/a",
      "hair_color": "n/a",
      "height": "167",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "75",
      "name": "C-3PO",
      "skin_color": "gold",
      "species": Array [
        "https://swapi.co/api/species/2/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/2/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "33BBY",
      "created": "2014-12-10T15:11:50.376000Z",
      "edited": "2014-12-20T21:17:50.311000Z",
      "eye_color": "red",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/5/",
        "https://swapi.co/api/films/4/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
        "https://swapi.co/api/films/7/",
      ],
      "gender": "n/a",
      "hair_color": "n/a",
      "height": "96",
      "homeworld": "https://swapi.co/api/planets/8/",
      "mass": "32",
      "name": "R2-D2",
      "skin_color": "white, blue",
      "species": Array [
        "https://swapi.co/api/species/2/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/3/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "41.9BBY",
      "created": "2014-12-10T15:18:20.704000Z",
      "edited": "2014-12-20T21:17:50.313000Z",
      "eye_color": "yellow",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "none",
      "height": "202",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "136",
      "name": "Darth Vader",
      "skin_color": "white",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [
        "https://swapi.co/api/starships/13/",
      ],
      "url": "https://swapi.co/api/people/4/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "19BBY",
      "created": "2014-12-10T15:20:09.791000Z",
      "edited": "2014-12-20T21:17:50.315000Z",
      "eye_color": "brown",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
        "https://swapi.co/api/films/7/",
      ],
      "gender": "female",
      "hair_color": "brown",
      "height": "150",
      "homeworld": "https://swapi.co/api/planets/2/",
      "mass": "49",
      "name": "Leia Organa",
      "skin_color": "light",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/5/",
      "vehicles": Array [
        "https://swapi.co/api/vehicles/30/",
      ],
    },
    Object {
      "birth_year": "52BBY",
      "created": "2014-12-10T15:52:14.024000Z",
      "edited": "2014-12-20T21:17:50.317000Z",
      "eye_color": "blue",
      "films": Array [
        "https://swapi.co/api/films/5/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "brown, grey",
      "height": "178",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "120",
      "name": "Owen Lars",
      "skin_color": "light",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/6/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "47BBY",
      "created": "2014-12-10T15:53:41.121000Z",
      "edited": "2014-12-20T21:17:50.319000Z",
      "eye_color": "blue",
      "films": Array [
        "https://swapi.co/api/films/5/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "female",
      "hair_color": "brown",
      "height": "165",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "75",
      "name": "Beru Whitesun lars",
      "skin_color": "light",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/7/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "unknown",
      "created": "2014-12-10T15:57:50.959000Z",
      "edited": "2014-12-20T21:17:50.321000Z",
      "eye_color": "red",
      "films": Array [
        "https://swapi.co/api/films/1/",
      ],
      "gender": "n/a",
      "hair_color": "n/a",
      "height": "97",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "32",
      "name": "R5-D4",
      "skin_color": "white, red",
      "species": Array [
        "https://swapi.co/api/species/2/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/8/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "24BBY",
      "created": "2014-12-10T15:59:50.509000Z",
      "edited": "2014-12-20T21:17:50.323000Z",
      "eye_color": "brown",
      "films": Array [
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "black",
      "height": "183",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "84",
      "name": "Biggs Darklighter",
      "skin_color": "light",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [
        "https://swapi.co/api/starships/12/",
      ],
      "url": "https://swapi.co/api/people/9/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "57BBY",
      "created": "2014-12-10T16:16:29.192000Z",
      "edited": "2014-12-20T21:17:50.325000Z",
      "eye_color": "blue-gray",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/5/",
        "https://swapi.co/api/films/4/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "auburn, white",
      "height": "182",
      "homeworld": "https://swapi.co/api/planets/20/",
      "mass": "77",
      "name": "Obi-Wan Kenobi",
      "skin_color": "fair",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [
        "https://swapi.co/api/starships/48/",
        "https://swapi.co/api/starships/59/",
        "https://swapi.co/api/starships/64/",
        "https://swapi.co/api/starships/65/",
        "https://swapi.co/api/starships/74/",
      ],
      "url": "https://swapi.co/api/people/10/",
      "vehicles": Array [
        "https://swapi.co/api/vehicles/38/",
      ],
    },
  ],
}
`;

exports[`should request luke 1`] = `
Object {
  "birth_year": "19BBY",
  "created": "2014-12-09T13:50:51.644000Z",
  "edited": "2014-12-20T21:17:56.891000Z",
  "eye_color": "blue",
  "films": Array [
    "https://swapi.co/api/films/2/",
    "https://swapi.co/api/films/6/",
    "https://swapi.co/api/films/3/",
    "https://swapi.co/api/films/1/",
    "https://swapi.co/api/films/7/",
  ],
  "gender": "male",
  "hair_color": "blond",
  "height": "172",
  "homeworld": "https://swapi.co/api/planets/1/",
  "mass": "77",
  "name": "Luke Skywalker",
  "skin_color": "fair",
  "species": Array [
    "https://swapi.co/api/species/1/",
  ],
  "starships": Array [
    "https://swapi.co/api/starships/12/",
    "https://swapi.co/api/starships/22/",
  ],
  "url": "https://swapi.co/api/people/1/",
  "vehicles": Array [
    "https://swapi.co/api/vehicles/14/",
    "https://swapi.co/api/vehicles/30/",
  ],
}
`;
